x/ibc-rate-limit/contracts/rate-limiter/src/sudo.rs

```diff
@@ -191,3 +191,26 @@ pub fn undo_send(deps: DepsMut, packet: Packet) -> Result<Response, ContractErro
         .add_attribute("denom", path.denom.to_string())
         .add_attribute("any_channel", (!any_trackers.is_empty()).to_string()))
 }
+
+
+/// helper function that is used to iterate over all existing rate limits automatically expiring flows for rules which have passed the end period
+pub fn rollover_expired_rate_limits(deps: DepsMut, env: Env) -> Result<(), ContractError> {
+    // possible alternative here is to not collect the iterator, and then use a dequeue or something similiar to track rate limit keys that need to be updated
+    for ((channel_id, denom), mut rules) in RATE_LIMIT_TRACKERS.range(deps.storage, None, None, cosmwasm_std::Order::Ascending).flatten().collect::<Vec<_>>() {
+        // avoid storage saves unless an actual rule was updated
+        let mut rule_updated = false;
+        rules.iter_mut().for_each(|rule| {
+            if rule.flow.is_expired(env.block.time) {
+                rule.flow.expire(env.block.time, rule.quota.duration);
+                // is this the correct way to reset the channel value??
```

Comment:
it isn't. The channel value is the current amount of tokens (the rate-limited denom) existing on osmosis (see https://github.com/osmosis-labs/osmosis/tree/main/x/ibc-rate-limit#notes-on-channel-value and https://github.com/osmosis-labs/osmosis/blob/ad31bfb16d8ecd476d3700079554b4e8842f9e7f/x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs#L115-L120)

This should be updated now that https://github.com/cosmos/ibc-go/issues/2664 has been merged.